% !TEX root = ./busty_transcription.tex
\begin{abstract}
The study of transcription remains one of the centerpieces of modern biology
with implications in settings from development to metabolism to evolution to
disease. Precision measurements using a host of different techniques including
fluorescence and sequencing readouts have raised the bar for what it means to
quantitatively understand transcriptional regulation. In particular our
understanding of the simplest genetic circuit is sufficiently refined both
experimentally and theoretically that it has become possible to carefully
discriminate between different conceptual pictures of how this regulatory system
works. This regulatory motif, originally posited by Jacob and Monod in the
1960s, consists of a single transcriptional repressor binding to a promoter site
and inhibiting transcription. In this paper, we show how seven distinct models
of this so-called simple-repression motif, based both on equilibrium
thermodynamic and chemical kinetic, can be used to derive the predicted levels
of gene expression and shed light on the often surprising past success of the
thermodynamic models. These different models are then invoked to confront a
variety of different data on mean, variance and full gene expression
distributions, illustrating the extent to which such models can and cannot be
distinguished, and suggesting a two-state model with a distribution of burst
sizes as the most potent of the seven for describing the simple-repression
motif.
\end{abstract}