% !TEX root = ./busty_transcription.tex
\section{Beyond Means in Gene Expression}
\label{sec:beyond_means}
 
In this section, our objective is to explore the same models considered in the
previous section, but now with reference to the question of how well they
describe the distribution of gene expression levels, with special reference to
the variance in these distributions. To that end, we repeat the same pattern as
in the previous section by examining the models one by one. In particular we
will focus on the Fano factor, defined as the variance/mean. This metric serves
as a powerful discriminatory tool to compare our different models to the null
model that the steady-state mRNA distribution must be Poisson, resulting a Fano
factor of one.

\subsection{Kinetic models for unregulated promoter noise}

Before we can tackle simple repression, we need an adequate phenomenological
model of constitutive expression. The literature abounds with options from which
we can choose, and we show several potential kinetic models for constitutive
promoters in Figure~\ref{fig2:constit_cartoons}(A). Let us consider
the suitability of each model for our purposes in turn.

\begin{figure}%[h!]
\centering
\includegraphics[width=\textwidth]{../../figures/main/fig02.pdf}
\caption{\textbf{Comparison of different models for noise in the constitutive promoter.}
(A) The left column depicts various plausible models for the dynamics of
constitutive promoters. In model (1), transcripts are produced in a Poisson
process~\cite{Sanchez2013, Jones2014}. Model (2) features explicit treatment of
RNAP binding/unbinding kinetics~\cite{Phillips2015a}. Model (3) is a more
detailed generalization of model (2), treating transcription initiation as a
multi-step process proceeding through closed and open
complexes~\cite{Mitarai2015}. Model (4) is somewhat analogous to (2) except with
the precise nature of active and inactive states left
ambiguous~\cite{Peccoud1995, Shahrezaei2008, Razo-Mejia2020}. Finally, model (5)
can be viewed as a certain limit of model (4) in which transcripts are produced
in bursts, and initiation of bursts is a Poisson process. 
% \mmnote{Haven't found model 5 formulated quite like this in the literature,
% usually burstiness is handled like model 4, but I should probably search a bit
% more.} 
The right column shows the Fano factor $\nu$ (variance/mean) for each model.
Note especially the crucial diagnostic: (2) and (3) have $\nu$ strictly below 1,
while only for (4) and (5) can $\nu$ exceed 1. Models with Fano factors $\le 1$
cannot produce the single-cell data observed in part (B) without introducing
additional assumptions and model complexity. (B) Data from \cite{Jones2014}.
Mean mRNA count vs. Fano factor (variance/mean) for different promoters as
determined with single-molecule mRNA Fluorescence \textit{in situ}
Hybridization. The colorbar indicates the predicted binding affinity of RNAP to
the promoter sequence as determined in \cite{Brewster2012}. Numbers serve for
cross comparison with data presented in Figure 3.}
\label{fig2:constit_cartoons}
\end{figure}

\subsubsection{Poisson Noise promoter}

The simplest model of constitutive expression that we can imagine is shown as
model 1 in Figure~\ref{fig2:constit_cartoons}(A) and assumes that transcripts
are produced as a Poisson process from a single promoter state. This is the
picture from Jones et.\ al.~\cite{Jones2014} that was used to interpret a
systematic study of gene expression noise over a series of promoters designed to
have different strengths, but no regulation. This model insists that the
``true'' steady-state mRNA distribution is Poisson, implying the Fano factor
$\nu$ must be 1. In~\cite{Jones2014}, the authors carefully attribute measured
deviations from Fano = 1 to intensity variability in fluorescence measurements,
gene copy number variation, and copy number fluctuations of the transcription
machinery, e.g., RNAP itself. In this picture, all the corrections to Poisson
behavior are derived as additive corrections to the Fano factor. This picture is
appealing in its simplicity, with only two parameters, the initiation rate $r$
and degradation rate $\gamma$. In other words, the model is not excessively
complex for the data at hand. But for many interesting questions, for instance
in the recent work~\cite{Razo-Mejia2020}, attributing all deviations from the
model to extrinsic noise sources, limits the kinds of predictions that can be
done. To make progress then we need a (slightly) more complex model than model 1
that would allow us to incorporate the non-Poissonian features of constitutive
promoters directly into a master equation formulation.

\subsubsection{Sub-Poissoninan noise promoters with RNAP escape}
A natural extension of the one-state promoter studied in the previous section is
to explicitly include an empty promoter state. This state allows for single RNAP
to bind and unbind from the promoter with rates $k_P^+$ and $k_P^-$,
respectively, before engaging in a transcriptional event. Once the RNAP
irreversibly escapes from the promoter to transcribe the mRNA, the promoter goes
back to this empty state. Models 2 and 3 from
Figure~\ref{fig2:constit_cartoons}(A) show two different versions of this
process. In model 2 the polymerase can directly engage in a transcription event
with rate $r$ after binding to the promoter, while model 3 includes an
irreversible step with rate $k_O$ that coarse-grains the known multi-step
process involved in transcriptional initiation~\cite{DeHaseth1998}.

Although physically intuitive given our understanding of how a single transcript
is produced, we are forced to discard both of these models on the basis that the
Fano factor for both of them is strictly $< 1$ as shown in
Figure~\ref{fig2:constit_cartoons}(A) (See Appendix~\ref{sec:non_bursty} for
detailed derivations). This is in disagreement with the experimental data
from~\cite{Jones2014}, reproduced in Figure~\ref{fig2:constit_cartoons}(B). All
of these unregulated promoters --which only differ in their promoter sequence--
show strictly super-Poissonian noise (Fano factor $> 1$). In fact, we suspect
\textit{any} model in which transcription proceeds through a multistep cycle
must necessarily have $\nu<1$. The intuitive argument compares the waiting time
distribution to traverse the cycle with the waiting time for a Poisson promoter
(model 1) with the same mean time. The latter is simply an exponential
distribution. The former is a convolution of multiple exponentials, and
intuitively the waiting time distribution for a multistep process should be more
peaked with a smaller fractional width than a single exponential with the same
mean. A less disperse waiting time distribution means transcription initations
are more uniformly distributed in time relative to a Poisson process. Hence the
distribution of mRNA over a population of cells should be less variable compared
to Poisson, giving $\nu<1$. (In Appendix~\ref{sec:non_bursty} we present a more
precise version of the intuitive arguments in this paragraph).




  




\subsubsection{Noise in a Two-State Promoter with ``Active'' and ``Inactive''
States}

Inspired by~\cite{Razo-Mejia2020}, we next revisit an analog of model 2 in
Figure~\ref{fig2:constit_cartoons}(A), but as with
the analogous models considered in Section~\ref{section_02_means},
the interpretation of the two
states is changed. Rather than explicitly viewing them as RNAP bound and
unbound, we view them as ``active'' and ``inactive,'' which are able and unable
to initiate transcripts, respectively. We are noncommittal as to the microscopic
details of these states.

One interpretation~\cite{Chong2014, Sevier2016, Sevier2018} for the active and
inactive states is that they represent the promoter's supercoiling state:
transitions to the inactive state are caused by accumulation of positive
supercoiling, which inhibits transcription, and transitions back to ``active''
are caused by gyrase or other topoisomerases relieving the supercoiling. This is
an interesting possibility because it would mean the timescale for promoter
state transitions is driven by topoisomerase kinetics, not by RNAP kinetics.
From in vitro measurements, the former are suggested to be of order
minutes~\cite{Chong2014}. Contrast this with model 2, where the state
transitions are assumed to be governed by RNAP, which, assuming a copy number
per cell of order $10^3$, has a diffusion-limited association rate $k_{on} \sim
10^2~\text{s}^{-1}$ to a target promoter. Combined with known $K_d$'s of order
$\mu$M, this gives an RNAP dissociation rate $k_{off}$ of order $10^2$~s$^{-1}$.
As we will show below, however, there are some lingering puzzles with
interpreting this supercoiling hypothesis, so we leave it as a speculative
hypothesis and refrain from assigning definite physical meaning to the two
states in this model.

Intuitively one might expect that, since transcripts are produced as a Poisson
process only when the promoter is in one of the two states in this model,
transcription initiations should now be ``bunched'' in time, in contrast to the
``anti-bunching'' of models 2 and 3 above. One might further guess that this
bunching would lead to super-Poissonian noise in the mRNA distribution over a
population of cells.  Indeed, as shown in Appendix~\ref{sec:non_bursty}, a
calculation of the Fano factor produces
\begin{equation}
\nu = 1 + \frac{r k^-}{(k^+ + k^- + \gamma)(k^+ + k^-)},
\label{eq:model4_fano}
\end{equation}
which is strictly greater than 1, verifying the above intuition. Note we have
dropped the $P$ label on the promoter switching rates to emphasize that these
very likely do not represent kinetics of RNAP itself. This calculation can also
be sidestepped by noting that the model is mathematically equivalent to the
simple repression model from~\cite{Jones2014}, with states and rates relabeled
and reinterpreted.

How does this model compare to model 1 above? In model 1, all non-Poisson
features of the mRNA distribution were handled as extrinsic corrections. By
contrast, here the 3 parameter model is used to fit the full mRNA distribution
as measured in mRNA FISH experiments. In essence, all variability in the mRNA
distribution is regarded as ``intrinsic,'' arising either from stochastic
initiation or from switching between the two coarse-grained promoter states. The
advantage of this approach is that it fits neatly into the master equation
picture, and the parameters thus inferred can be used as input for more
complicated models with regulation by transcription factors.

While this seems promising, there is a major drawback for our purposes which was
already uncovered by the authors of~\cite{Razo-Mejia2020}: the statistical
inference problem is nonidentifiable, in the sense described in Section 4.3
of~\cite{Gelman2013}. What this means is that it is impossible to infer the
parameters $r$ and $k^-$ from the single-cell mRNA counts data
of~\cite{Jones2014} (as shown in Fig.~S2 of~\cite{Razo-Mejia2020}). Rather, only
the ratio $r/k^-$ could be inferred. In that work, the problem was worked around
with an informative prior on the ratio $k^-/k^+$. That approach is unlikely to
work here, as, recall, our entire goal in modeling constitutive expression is to
use it as the basis for a yet more complicated model, when we add on repression.
But adding more complexity to a model that is already poorly identified is a
fool's errand, so we will explore one more potential model.

\subsubsection{Noise Model for One-State Promoter with Explicit Bursts}

The final model we consider is inspired by the failure mode of model 4. The key
observation above was that, as found in~\cite{Razo-Mejia2020}, only two
parameters, $k^+$ and the ratio $r/k^-$, could be directly inferred from the
single-cell mRNA data from~\cite{Jones2014}. So let us take this seriously and
imagine a model where these are the only two model parameters. What would this
model look like?

To develop some intuition, consider model 4 in the limit $k^+ \ll k^- \lesssim
r$, which is roughly satisfied by the parameters inferred
in~\cite{Razo-Mejia2020}. In this limit, the system spends the majority of its
time in the inactive state, occasionally becoming active and making a burst of
transcripts. This should call to mind the well-known phenomenon of
transcriptional bursting, as observed in,
e.g.,~\cite{Golding2005,Chong2014,Senecal2014,Sevier2016,Sepulveda2016}. Let us
make this correspondence more precise. The mean dwell time in the active state
is $1/k^-$. While in this state, transcripts are produced at a rate $r$ per unit
time. So on average, $r/k^-$ transcripts are produced before the system switches
to the inactive state. Once in the inactive state, the system dwells there for
an average time $1/k^+$ before returning to the active state and repeating the
process. $r/k^-$ resembles an average burst size, and $1/k^+$ resembles the time
interval between burst events. More precisely, $1/k^+$ is the mean time between
the end of one burst and the start of the next, whereas $1/k^+ + 1/k^-$ would be
the mean interval between the start of two successive burst events, but in the
limit $k^+ \ll k^-$, $1/k^+ + 1/k^- \approx 1/k^+$. Note that this limit ensures
that the waiting time between bursts is approximately exponentially distributed,
with mean set by the only timescale left in the problem, $1/k^+$. If instead it
were the case that $k^+ \sim k^-$, then the waiting time $1/k^+ + 1/k^-$ would
have a peaked distribution, but this does not appear to be the case for any of
datasets from~\cite{Jones2014}.

Let us now verify this intuition with a precise derivation to check that $r/k^-$
is in fact the mean burst size and to obtain the full burst size distribution.
Consider first a constant, known dwell time $T$ in the active state. Transcripts
are produced at a rate $r$ per unit time, so the number of transcripts $n$
produced during $T$ is Poisson distributed with mean $rT$, i.e.,
\begin{equation}
    P(n\mid T) = \frac{(rT)^n}{n!} \exp(-rT).
\end{equation}
Since the dwell time $T$ is unobservable from single molecule mRNA counts, we
actually want $P(n)$, the dwell time distribution with no conditioning on $T$.
Basic rules of probability theory tell us we can write $P(n)$ in terms of
$P(n\mid T)$ as
\begin{equation}
    P(n) =\int_0^\infty P(n\mid T) P(T) dT.
\end{equation}
But we know the dwell time distribution $P(T)$, which is exponentially
distributed according to
\begin{equation}
    P(T) = k^- \exp(-T k^-),
\end{equation}
so $P(n)$ can be written as
\begin{equation}
    P(n) = k^- \frac{r^n}{n!}
            \int_0^\infty T^n\exp[-(r + k^-)T]\,dT.
\end{equation}
A standard integral table shows $\int_0^\infty x^n e^{-ax}\,dx = n!/a^{n+1}$, so
\begin{equation}
    P(n) = \frac{k^- r^n}{(k^- + r)^{n+1}}
        = \frac{k^-}{k^- + r}
            \left(\frac{r}{k^- + r}\right)^n
        = \frac{k^-}{k^- + r}
            \left(1 - \frac{k^-}{k^- + r}\right)^n,
\end{equation}
which is exactly the geometric distribution with standard parameter
$\theta\equiv k^-/(k^- + r)$ and domain $n \in \{0, 1, 2, \dots\}$.
The mean of the geometric distribution, with this convention, is
\begin{equation}
\langle n\rangle = \frac{1 - \theta}{\theta}
        = \left(1 - \frac{k^-}{k^- + r}\right)
                    \frac{k^- + r}{k^-}
        = \frac{r}{k^-},
\end{equation}
exactly as we guessed intuitively above.

So in taking the limit $r,k^-\rightarrow\infty$, $r/k^-\equiv b$, we obtain a
model which effectively has only a single promoter state, which initiates bursts
at rate $k^+$ (transitions to the active state, in the model 4 picture). The
master equation for mRNA copy number $m$ as derived in
Appendix~\ref{sec:non_bursty} takes the form
\begin{equation}
\begin{split}
{d \over dt}p(m,t) = & (m+1)\gamma p(m+1,t) - m\gamma p(m,t) \\
        &+ \sum_{m'=0}^{m-1} k_i p(m',t) \text{Geom}(m-m';b)
         - \sum_{m'=m+1}^\infty k_i p(m,t) \text{Geom}(m'-m;b),
\end{split}
\end{equation}
where we use $k_i$ to denote the burst initiation rate, $\text{Geom}(n;b)$ is
the geometric distribution with mean~$b$, i.e., $\text{Geom}(n;b) =
\frac{1}{1+b}\left(\frac{b}{1+b}\right)^n$ (with domain over nonnegative
integers as above). The first two terms are the usual mRNA degradation terms.
The third term enumerates all ways the system can produce a burst of transcripts
and arrive at copy number $m$, given that it had copy number $m'$ before the
burst. The fourth term allows the system to start with copy number $m$, then
produce a burst and end with copy number $m'$. In fact this last sum has trivial
$m'$ dependence and simply enforces normalization of the geometric distribution.
Carrying it out we have
\begin{equation}
\begin{split}
{d \over dt} p(m,t) = & (m+1)\gamma p(m+1,t) - m\gamma p(m,t) \\
        &+ \sum_{m'=0}^{m-1} k_i p(m',t) \text{Geom}(m-m';b)
            - k_i p(m,t),
\end{split}
\end{equation}
We direct readers again to Appendix~\ref{sec:gen_fcn_appdx} for further details.
This improves on model 4 in that now the parameters are easily inferred, as we
will see later, and have clean interpretations. The non-Poissonian features are
attributed to the empirically well-established phenomenological picture of
bursty transcription.

The big approximation in going from model 4 to 5 is that a burst is produced
instantaneously rather than over a finite time. If the true burst duration is
not short compared to transcription factor kinetic timescales, this could be a
problem in that mean burst size in the presence and absence of repressors could
change, rendering parameter inferences from the constitutive case inappropriate.
Let us make some simple estimates of this.

Consider the time delay between the first and final RNAPs in a burst initiating
transcription (\textit{not} the time to complete transcripts, which potentially
could be much longer.) If this timescale is short compared to the typical search
timescale of transcription factors, then all is well. The estimates from
deHaseth et.\ al.~\cite{DeHaseth1998} put RNAP's diffusion-limited on rate
around $\sim\text{few}\times10^{-2}~\text{nM}^{-1}~\text{s}^{-1}$ and polymerase
loading as high as $1~\text{s}^{-1}$. Then for reasonable burst sizes of $<10$,
it is reasonable to guess that bursts might finish initiating on a timescale of
tens of seconds or less (with another 30-60 sec to finish elongation, but that
does not matter here). A transcription factor with typical copy number of order
10 (or less) would have a diffusion-limited association rate of order
$(10~\text{sec})^{-1}$ \cite{Hammar2014}. Higher copy number TFs tend to have
many binding sites over the genome, which should serve to pull them out of
circulation and keep their effective association rates from rising too large.
Therefore, there is \textit{perhaps} a timescale separation possible between
transcription factor association rates and burst durations, but this assumption
could very well break down, so we will have to keep it in mind when we infer
repressor rates from the Jones et.\ al.\ single-cell mRNA counts data
later~\cite{Jones2014}.

In reflecting on these 5 models, the reader may feel that exploring a multitude
of potential models just to return to a very minimal phenomenological model of
bursty transcription may seem highly pedantic. But the purpose of the exercise
was to examine a host of models from the literature and understand why they are
insufficient, one way or another, for our purposes. Along the way we have
learned that the detailed kinetics of RNAP binding and initiating transcription
are probably irrelevant for setting the population distribution of mRNA. The
timescales are simply too fast, and as we will see later in
Figures~\ref{fig:constit_post_full} and~\ref{fig4:repressed_post_full},
the noise seems to be governed by slower timescales.
Perhaps in hindsight this is not surprising: intuitively, the degradation rate
$\gamma$ sets the fundamental timescale for mRNA dynamics, and any other
processes that substantially modulate the mRNA distribution should not differ
from $\gamma$ by orders of magnitude.