% !TEX root = ./busty_transcription.tex
\section{Discussion and future work}

The study of gene expression is one of the dominant themes of modern
biology, made all the more urgent by the dizzying pace at which
genomes are being sequenced.  But there is a troubling Achilles heel
buried in all of that genomic data and that is our inability to find
and interpret regulatory sequence.  In many cases, this is not
possible even qualitatively, let alone the possibility of quantitative
dissection of the regulatory parts of genomes in a predictive fashion.
Other recent work has tackled the challenge of finding and
annotating the regulatory part of genomes.  Once we have determined
the regulatory part of the genome, we are then faced with the
next class of questions which are sharpened by formulating
them in mathematical terms, namely, what are the input-output
properties of these regulatory circuits and what knobs control them?

The present work has tackled that question in the context of
the first regulatory architecture hypothesized in the molecular
biology era, namely, the repressor-operator model of Jacob and
Monod.  Regulation in that architecture is the result of a 
competition between a repressor which inhibits transcription
and RNAP polymerase which undertakes it.  Through
the labors of generations of geneticists, molecular biologists
and biochemists, an overwhelming amount of information
and insight has been garnered into this simple regulatory
motif, licensing it as what one might call the ``hydrogen atom''
of regulatory biology.   It is from that perspective that the present
paper explores the extent to which some of the different models
that have been articulated to describe that motif allow
us to understand both the average level of gene expression
found in a population of cells as well as the noise in that
expression found in that population as would be reported in
a FISH experiment, for example.

Our key insights can be summarized as follows.  First, as shown in Figure
Figure~\ref{fig1:means_cartoons}, the mean expression in the
simple repression architecture is captured by a master curve
in which the action of repressor and the details of the polymerase
interaction with the promoter appear separately and additively
in an effective free energy.  Interestingly, as has been shown
elsewhere in the context of the Monod-Wyman-Changeux model,
these kinds of coarse-graining results are an exact mathematical
result and do not constitute hopeful approximations or biological
naivete.  To further dissect the relative merits of the different models,
we must appeal to higher moments of the gene expression probability
distribution.  To that end, our second set of insights focus on gene
expression noise, where it is seen that a treatment of
the constitutive promoter already reveals that some models have Fano
factors that are less than one, at odds with any and all experimental
data that we are aware of.   This refinement allows us to determine
parameters for the constitutive promoter that can then be used
to attack gene expression noise in the context of
the simple repression architecture.




\mmnote{Do all my approximations work simply because galK happens to have a low
duty cycle of supercoiling, $\beta/\alpha$ in Chong2014 picture? A potentially
interesting test: put in the dozen or so Hernan constructs at some other locus,
preferably one of the ones Chong identified as having a large duty cycle (large
$\beta/\alpha$), i.e., the opposite limit of galK. Then just measure
fold-changes. You don't even need to do FISH, although that would be even
better. If their model is right, $\rho$ should be approximately set by the duty
cycle of supercoiling, NOT by RNAP kinetics.
\textit{This is a NICE prediction for future work.} \textit{cspE}
is claimed nonessential, prediction is you drop HG reporter constructs in there
and the inferred binding energies of all lacI operators shift the same amount,
by about $\ln(3)\sim 1kT$. Not huge, but detectable, although might be
confounding effects from moving to a new locus. With FISH measurements, $k^+$
and $k^-$ in Fig 2, model 4 should now be separately inferrable if Chong2014
measurements of $\beta/\alpha$ are correct. This would require an obnoxious
amount of cloning, but once you did that, the measurements would be same old
stuff that RP lab does.}

It is interesting to speculate what microscopic details are being
coarse-grained by our burst rate and burst size in model 5.
Chromosomal locus is one possible influence we have not addressed
in this work, as all the FISH data from~\cite{Jones2014} that we
considered was from a construct integrated at the \textit{galK} locus.
The results of~\cite{Chong2014} indicate that
transcription-induced supercoiling contributes substantially in
driving transcriptional bursting, and furthermore, their Figure~7
suggests that the parameters describing the rate, duration, and
size of bursts vary substantially for transcription from
different genomic loci.
Although the authors of~\cite{Englaender2017} do not address
noise, they note enormous differences in mean expression levels
when an identical construct is integrated at different genomic loci.
The authors of~\cite{Engl2020} attribute noise and burstiness in
their FISH data to the influence of different sigma factors,
which is a reasonable conclusion from their data.
Could the difference also be due to the different chromosomal
locations of the two operons?

It seems clear therefore that chromosomal locus plays a major
role in mean expression levels as well as noise. What features of
different loci are and are not important? Sigma factors could
certainly play a part, but it seems many other mechanisms are at
play: supercoiling, and more generally 3D organization of the
chromosome, read-through of neighboring genes, and surely some
fascinating unknown unknowns.
Could our preferred coarse-grained model capture the variability
across different loci? If so, and we were to repeat the parameter
inference as done in this work, is there a simple theoretical model
we could build to understand the resulting parameters?

A direct \mmnote{if cloning intensive} future experiment would be
to design a payload with a promoter driving a FISH reporter,
integrate that payload at a variety of loci scattered around the genome,
and carry out FISH measurements of the resulting strains.
\mmnote{
Did Brewster already do this? I remember he integrated arrays of
competitor binding sites all over the genome, but did he put
lacUV5 + reporter scattered around?}
Obviously integration loci would need to be carefully chosen so
that one could plausibly hope the deletion and integration did
not disturb the cell's physiology too badly.