% !TEX root = ./busty_transcription.tex
\section{Discussion and future work}

The study of gene expression is one of the dominant themes of modern biology,
made all the more urgent by the dizzying pace at which genomes are being
sequenced. But there is a troubling Achilles heel buried in all of that genomic
data, which is our inability to find and interpret regulatory sequence. In
many cases, this is not possible even qualitatively, let alone the possibility
of quantitative dissection of the regulatory parts of genomes in a predictive
fashion. Other recent work has tackled the challenge of finding and annotating
the regulatory part of genomes \cite{Belliveau2018, Ireland2020}. Once we have
determined the architecture of the regulatory part of the genome, we are then
faced with the next class of questions which are sharpened by formulating them
in mathematical terms, namely, what are the input-output properties of these
regulatory circuits and what knobs control them?

The present work has tackled that question in the context of the first
regulatory architecture hypothesized in the molecular biology era, namely, the
repressor-operator model of Jacob and Monod~\cite{Jacob1961}. Regulation in that
architecture is the result of a competition between a repressor which inhibits
transcription and RNAP polymerase which undertakes it. Through the labors of
generations of geneticists, molecular biologists and biochemists, an
overwhelming amount of information and insight has been garnered into this
simple regulatory motif, licensing it as what one might call the ``hydrogen
atom'' of regulatory biology. It is from that perspective that the present paper
explores the extent to which some of the different models that have been
articulated to describe that motif allow us to understand both the average level
of gene expression found in a population of cells, the intrinsic cell-to-cell
variability, and the full gene expression distribution found in such a population
as would be reported in a single molecule mRNA Fluorescence \textit{in situ}
Hybridization experiment, for example.

Our key insights can be summarized as follows. First, as shown in
Figure~\ref{fig1:means_cartoons}, the mean expression in the simple repression
architecture is captured by a master curve in which the action of repressor and
the details of the RNAP interaction with the promoter appear separately and
additively in an effective free energy. Interestingly, as has been shown
elsewhere in the context of the Monod-Wyman-Changeux model, these kinds of
coarse-graining results are an exact mathematical result and do not constitute
hopeful approximations or biological naivete \cite{Razo-Mejia2018, Chure2019}.
To further dissect the relative merits of the different models, we must appeal
to higher moments of the gene expression probability distribution. To that end,
our second set of insights focus on gene expression noise, where it is seen that
a treatment of the constitutive promoter already reveals that some models have
Fano factors (variance/mean) that are less than one, at odds with any and all
experimental data that we are aware of~\cite{So2011, Jones2014}. This theoretical
result allows us to directly discard a subset of the models (models 1-3 in
Figure~\ref{fig2:constit_cartoons}(A)) since they cannot be reconciled with
experimental observations. The two remaining models (models 4 and 5 in
Figure~\ref{fig2:constit_cartoons}) appear to contain enough microscopic realism
to be able to reproduce the data. A previous exploration of model 4 demonstrated
the ``sloppy''~\cite{Transtrum2015} nature of the model
in which data on single-cell mRNA counts alone
cannot constrain the value of all parameters simultaneously
\cite{Razo-Mejia2020}. Here we demonstrate that the proposed one-state bursty
promoter model (model 5 in Figure~\ref{fig2:constit_cartoons}(A)) emerges as a
limit of the commonly used two-state promoter model \cite{Peccoud1995,
Shahrezaei2008, So2011, Sanchez2013, Jones2014}. We put the idea to the test
that this level of coarse-graining is rich enough to reproduce previous
experimental observations. In particular we perform Bayesian inference to
determine the two parameters describing the full steady-state mRNA distribution,
finding that the model is able to provide a quantitative description of a
plethora of promoter sequences with different mean levels of expression and
noise.

With the results of the constitutive promoter in hand, we then fix the
parameters associated with this class of promoters and use them as input for
evaluating the noise in gene expression for the simple repression motif itself.
This allows us to provide a single overarching analysis of both the constitutive
and simple repression architectures using one simple model and corresponding set
of self-consistent parameters, demonstrating not only a predictive framework,
but also reconciling the equilibrium and non-equilibrium views of the same
simple repression constructs. More specifically, we obtained values for the
transcription factor association and dissociation rates by performing Bayesian
inference on the full mRNA distribution for data obtained from simple-repression
promoters with varying number of transcription factors per cell and affinity of
such transcription factors for the binding site. The free energy value obtained
from these kinetic rates -- computed as the log ratio of the rates -- agrees
with previous inferences performed only from mean gene expression measurements,
that assumed an equilibrium rather than a kinetic
framework~\cite{Garcia2011a, Razo-Mejia2018}.

It is interesting to speculate what microscopic details are being coarse-grained
by our burst rate and burst size in Figure~\ref{fig2:constit_cartoons}, model 5.
Chromosomal locus is one possible influence we have not addressed in this work,
as all the single-molecule mRNA data from~\cite{Jones2014} that we considered
was from a construct integrated at the \textit{galK} locus. The results
of~\cite{Chong2014} indicate that transcription-induced supercoiling contributes
substantially in driving transcriptional bursting, and furthermore, their
Figure~7 suggests that the parameters describing the rate, duration, and size of
bursts vary substantially for transcription from different genomic loci.
Although the authors of~\cite{Englaender2017} do not address noise, they note
enormous differences in mean expression levels when an identical construct is
integrated at different genomic loci. The authors of~\cite{Engl2020} attribute
noise and burstiness in their single-molecule mRNA data to the influence of
different sigma factors, which is a reasonable conclusion from their data. Could
the difference also be due to the different chromosomal locations of the two
operons?
What features of different loci are and are not important?
Could our preferred coarse-grained model capture the variability across
different loci? If so, and we were to repeat the parameter inference
as done in this work, is there a simple theoretical model we could build
to understand the resulting parameters?

In summary, this work took up the challenge of exploring the extent to which a
single specific mechanistic model of the simple-repression regulatory
architecture suffices to explain the broad sweep of experimental data for this
system. Pioneering early experimental efforts from the M\"{u}ller-Hill lab
established the simple-repression motif as an arena for the quantitative
dissection of regulatory response in bacteria, with similar beautiful work
emerging in examples such as the \textit{ara} and \textit{gal} operons as
well~\cite{Dunn1984b, Oehler1990, Weickert1993, Oehler1994, Schleif2000,
Semsey2002, SwintKruse2009}. In light of a new generation of precision
measurements on these systems, the definition of what it means to understand
them can now be formulated as a rigorous quantitative question. In particular,
we believe understanding of the simple repression motif has 
advanced sufficiently that the design of new
versions of the architecture is now possible,
based upon predictions about how repressor copy
number and DNA binding site strength control expression. In our view, the next
step in the progression is to first perform similar rigorous analyses of the
fundamental ``basis set'' of regulatory architectures. Natural follow-ups to this
work are explorations of motifs such as simple activation that is
regulated by a single activator binding site, and the repressor-activator
architecture, mediated by the binding of both a single activator and a single
repressor, and beyond. With the individual input-output functions in hand,
similar quantitative dissections including the rigorous analysis of their tuning
parameters can be undertaken for the ``basis set'' of full gene-regulatory
networks such as switches, feed-forward architectures and oscillators for
example, building upon the recent impressive bonanza of efforts from systems
biologists and synthetic biologists~\cite{Milo2002, Alon2007}.